packages/eas-cli/src/commands/update/index.ts

```diff
@@ -586,15 +613,43 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return [transformRuntimeVersions(exp, platforms), null];
+  } catch (error: any) {
+    if (nonInteractive) {
+      Errors.error(error, { exit: 1 });
+    }
+
+    ora().start().fail(error.message);
+
+    const runConfig = await selectAsync(
+      `Do you want us to run automatic ${chalk.bold(
+        'app.json'
+      )} configuration for EAS Update for you?`,
+      [
+        { title: 'Yes', value: true },
+        {
+          title: 'No, I will set the runtime version manually (EAS CLI exits)',
+          value: false,
+        },
+      ]
+    );
+
+    if (!runConfig) {
+      Errors.exit(1);
+    }
+
+    const { configureAppJSONForEASUpdateAsync } = await import('./configure');
```

Comment:
Why do you need the dynamic import?

Reply:
Performance, it's the common pattern in Expo CLI.

Reply:
It's not Expo CLI, please don't use dynamic import in this repo.

Reply:
Also, this is "need" IHMO, since this path will be triggered only on non-setup or invalid setup projects which make the percentile of `update` calls, and with static import we will always load whole configure step, which affects the command stat-up and TTI time.

Reply:
Can you prepare some benchmarks on what is the performance gain? I'd say it is negligible.

Reply:
I'm not saying about benchmarks in general. I'm saying that I want to see benchmarks for this particular case. As Wojtek pointed out, there is no real gain here as probably all other modules imported in the `configure` module are imported anyway. 
The problem with the code that you proposed is that using dynamic imports in **EAS CLI** would be a precedence. In this case, I'm afraid, it's not at all justified.

Reply:
Also unless ocliff is doing lazy loading per command there might be no impact on performance at all regardless of the size of the imported part.

Reply:
https://github.com/expo/eas-cli/pull/1427#discussion_r991500606

https://github.com/expo/eas-cli/blob/024c699637f46914d0ba9766f321faaf5fc4efb4/packages/eas-cli/src/metadata/config.ts#L23

Reply:
This line is reading the config from the user's project. It's not the same. 
In you're code you're importing internal modules.

Reply:
> there is no real gain here as probably all other modules imported in the configure module are imported anyway.

There are no other `configure` imports or usage in the `update` command, and imports are loaded per command on run, so I think that the sentence above is incorrect.

Reply:
Just compare the imports in those files.
<img width="900" alt="Screenshot 2022-10-12 at 12 53 31" src="https://user-images.githubusercontent.com/5256730/195325186-2ca5a335-dcc8-462d-9733-e3caa15b464c.png">
<img width="1065" alt="Screenshot 2022-10-12 at 12 53 17" src="https://user-images.githubusercontent.com/5256730/195325205-d1111229-c58e-4407-bbc0-fcb89e7bee69.png">

I won't believe there's any performance gain until I see data :)

Reply:
Now that this is just a normal function being imported I think it it makes sense to just use a normal import (we do this for pretty much all business logic functions). 

The key though is that we should move the `configureAppJSONForEASUpdateAsync` out of the command module and into its own module. If it's in the command module, a lot of extra stuff pertaining to commands and command static stuff is loaded and I'd be willing to bet that the dynamic import is more optimal since it wouldn't be loaded in most cases. If it's in its own module, then none of the command-related stuff will be loaded and the overhead of loading it normally is negligible. If it's in its own module, it is pretty much the same as any other function being imported in any command module.